ArmRobot/Assets/Scripts/DomainRandomization.cs

```diff
@@ -0,0 +1,218 @@
+        List<GameObject> listOfCylinders = CreateRandomListCylinders();
+        
+        // Then we create the spheres 
+        List<GameObject> listOfSpheres = CreateRandomListSpheres();
+        
+        // Then we create the list gathering all the objects which are on the table that we have created 
+        List<GameObject> listOfObjectsTable = ConcatenateListOfGameObjects(listOfCylinders, listOfSpheres);
+
+        // we also create a list of lights objects but these objects will not be part of the list of Objects 
+        List<GameObject> listOfLights = CreateRandomListLights();
+
+        return listOfObjectsTable;
+    }
+    
+    
+    public void DomainRandomizationScene()
+    {
+        // we create a list that will contains all the objects we have already moved 
+        List<GameObject> listOfAlreadyMovedObjects = new List<GameObject>();
+
+        // Then we start moving the objects and changing the color 
+        // move cube
+        TablePositionRandomizer tablePositionRandomizer_cube = cube.GetComponent<TablePositionRandomizer>();
+        tablePositionRandomizer_cube.Move();
+        
+        
+        // then we change its color 
+        ColorRandomizer colorRandomizercube = cube.GetComponent<ColorRandomizer>();
+        colorRandomizercube.ChangeColor();
+
+        listOfAlreadyMovedObjects.Add(cube);
+
+        // We destroy the older objects 
+        DestroyObjects();
+
+        // Create and move the new objects 
+        // We create the new ojects 
+        List<GameObject> listOfObjectsTable = InitializationObjects();
+
+        // then we move them
+        // we iterate through the listOfObjects and move them one by one 
+        foreach (GameObject gameobject in listOfObjectsTable) {
+            RandomizerPositionObject tablePositionRandomizerObject = gameobject.GetComponent<RandomizerPositionObject>();
+            tablePositionRandomizerObject.Move(listOfAlreadyMovedObjects);
+            listOfAlreadyMovedObjects.Add(gameobject); // we add the gameobject to the already moved objects 
+        }
+        
+        
+        // move robot 
+        MoveRobot(rotationAngle);
+        // then we change its color and the color of all its children
+        foreach (Transform child in robot.transform){
+            ColorRandomizer colorRandomizerChildRobot = child.gameObject.GetComponent<ColorRandomizer>();
+            colorRandomizerChildRobot.ChangeColor();
+        } 
+        
+        // move the camera
+        GameObject camera = GameObject.Find("VisionCamera");
+        RandomizerPositionCamera randomizedPositionCamera = camera.GetComponent<RandomizerPositionCamera>();
+        randomizedPositionCamera.Move();
+
+        // change the color of the table 
+        ColorRandomizer colorRandomizerTable = table.GetComponent<ColorRandomizer>();
+        colorRandomizerTable.ChangeColor();
+
+        // change the Directionalight
+        GameObject directionLight = GameObject.Find("DirectionalLight");
```

Comment:
I get an error when I run the project because of this line. It is named `Directional Light` not `DirectionalLight`. Because of that, it is not actually finding the component.

Reply:
Yes sorry I just found out this error